### PR TITLE
Feat: add page title

### LINKED ISSUE #49: PageTitle component
Possible names:

- PageTitle
- ScreenTitle
- ViewTitle

I expect that a lot of sites will wind up having a SiteTitle component or variable in some form or another, so we'd probably want to avoid just plain old Title as a name since it could potentially be confusing.

This should use the PageTitle styles from the `mixins.js` file and more or less follow Heading conventions with the following exceptions:

- Should use an h1 (probably with no possibility of a level override?)
- Only one per screen/page/view allowed.
- Do we want to use Sentence case here or switch to Title Case Instead? Title case could actually be nice, but I have mixed feelings there.
- Can allow for more expressive visual styling (gradients inside the text, funky effects, weird placement, etc).

### PR DESCRIPTION
**Pull request by tofumatt**
Closes #49.

This doesn't _use_ `PageTitle` anywhere on our site yet but I'd like to integrate it into things after I get it merged and tweak the markdown output further in a separate PR. I think doing that after we have lists is best, because then I'll be able to use a lot of our base components for the style guide.

### CODE REVIEW

File: styleguide/docs/components.md [hunk 1]
```diff
@@ -8,7 +8,7 @@ const components = Object.keys(loadComponents).map((name) => {
 
 	return (
 		<li key={name}>
-			<a href={`#/Function/Components/${normalisedName}`}>
+			<a href={`#/🛠%20Function/Components/${normalisedName}`}>
```
Comment (tofumatt) [inline styleguide/docs/components.md]: This is unrelated; you can ignore it 😄

File: src/components/PageTitle/README.md [hunk 2]
```diff
@@ -0,0 +1,43 @@
...
+
+You can think of `<PageTitle>` as a combination `<title>` and `<h1>` tag; it is used to describe the content of the entire page/screen available to the user, and should not change unless the user navigates away from the current view. **Every page you create should have one and only one `<PageTitle>` component.**
+
+```jsx
+<PageTitle>My Homepage</PageTitle>
+```
+
+## Usage
+
+Use a `<PageTitle>` to introduce the user to the entirety of the content available in the current view. On a blog, for instance, your index view might use: `<PageTitle>My Posts</PageTitle>` and an individual post would use its post title: `<PageTitle>{post.title}</PageTitle>`.
```
Comment (sarahmonster) [inline src/components/PageTitle/README.md]: I'm rearranging this to a "Technical Implementation" section!

File: src/components/PageTitle/README.md [hunk 3]
```diff
@@ -0,0 +1,43 @@
...
+
+<!-- TODO for @tofumatt: implement this feature. -->
+When using `<DocumentTitle>`, which uses `react-helmet`, the content inside `<PageTitle>` will be inserted into your page's `<title>` tag. If your `<PageTitle>` contains a link or other content aside from plain text, you should use the `documentTitle` prop to set the content that will be used by `<DocumentTitle>`:
+
+```jsx inside Markdown
```
Comment (sarahmonster) [inline src/components/PageTitle/README.md]: This should go under "Variations" but before I move it ... why would you want to link an h1/page title ever?

**Comment by tofumatt** [comment]
Sometimes blogs and the like will provide an easy way for users to get a link to the current page from the header. I guess it helps non-technical users get the link? Maybe we don’t want to encourage that though; I’m fine with not!

**Comment by sarahmonster** [comment]
Hm yeah, in my experience building way too many WordPress themes, typically one uses a link only for post titles on _archive_ pages (where they are h2 level), and no link on single post pages (where they have an h1 level). 

Like so: https://github.com/Automattic/_s/blob/28f0a5b4e56a5dc3cba94749a208eb478fe224d4/template-parts/content.php#L15

File: src/components/PageTitle/README.md [hunk 4]
```diff
@@ -0,0 +1,50 @@
+A Page Title is a short line of text that describes the current page or screen. Every page or screen in your site or app should have **one** `<PageTitle>` component: no more, no less.
```
Comment (tofumatt) [inline src/components/PageTitle/README.md]: Much better 👍

File: src/components/PageTitle/README.md [hunk 5]
```diff
@@ -0,0 +1,50 @@
...
+## Appearance
+
+- Use a consistent style rules for `<PageTitle>`s across your site, even if you're doing creative art direction.
+- Since a `<PageTitle>` is highest in hierarchy, it should be larger or have more emphasis than other `<Heading>`s.
+- `<PageTitle>`s can use styling similar to `<Heading>`s, but doesn't need to. Since they are the highest in the hierarchy and only appear once per view, you can feel free to get a little creative with them. Highlight your brand, apply funky styles like gradients or skews, use hand-lettering effects, or art-direct each one with different typography and special styling.
+
+## Voice & Tone
+
+- Keep the title short and descriptive of the page's content.
+- `<PageTitle>` can use Sentence case or Title Case, depending on your personal tastes and the formality of your site or app. Whichever you pick, be consistent throughout!
```
Comment (tofumatt) [inline src/components/PageTitle/README.md]:
```suggestion
- `<PageTitle>` can use Sentence case or Title Case, depending on your personal tastes and the formality of your site or app. Whichever you pick: be consistent throughout!
```

Comment (sarahmonster) [reply src/components/PageTitle/README.md]: You're wrong. I'm shipping this.
